### PR TITLE
Topics

### PR DESCRIPTION
**Pull request by heimrih**
Topics with posts

### CODE REVIEW

**Comment by heimrih** [comment]
ready

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -4,7 +4,15 @@ class PostsController < ApplicationController
   before_action :correct_user, only: %i(edit update)
 
   def index
-    @posts = Post.order(created_at: :desc).page(params[:page]).per Settings.posts.page.max
+    @topics = Topic.all
```
Comment (vanvtt-0952) [inline app/controllers/posts_controller.rb]: add scope sort
example
- in Topic model file
`scope :lastest, ->{order created_at: :desc}`

- in here
change code ` @topics = Topic.all` to ` @topics = Topic.lastest`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -4,7 +4,15 @@ class PostsController < ApplicationController
...
+
+    top = params[:top]
+
+    if !top.nil?
```
Comment (vanvtt-0952) [inline app/controllers/posts_controller.rb]: `!top.nil?` same result `top.present?`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -4,7 +4,15 @@ class PostsController < ApplicationController
...
+      @posts = Post.where(:topic_id => top).page(params[:page]).per Settings.posts.page.max
+    else
+      @posts = Post.all.order(created_at: :desc).page(params[:page]).per Settings.posts.page.max
+    end
```
Comment (vanvtt-0952) [inline app/controllers/posts_controller.rb]: I think you should change code from 11 line - 15 line to
- in Post model file
add code:
`scope :lastest, ->{order created_at: :desc}`
`scope :by_top, ->(top_id){where top_id: top_id if top_id.present?}`
- in here, change code from 11 line to 15 line:
@posts = Post.by_top(top).lastest.page(params[:page]).per Settings.posts.page.max

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

**Comment by heimrih** [comment]
ready

**Pull request merged**